x/did/handler.go

```diff
@@ -27,14 +27,38 @@ func handleMsgAddDidDoc(ctx sdk.Context, k keeper.Keeper, msg types.MsgAddDid) s
 	if err != nil {
 		return err.Result()
 	}
-	return sdk.Result{}
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			types.EventTypeAddDidDoc,
+			sdk.NewAttribute(types.AttributeKeyDid, msg.Did),
+			sdk.NewAttribute(types.AttributeKeyPubKey, msg.PubKey),
+		),
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+		),
+	})
+	return sdk.Result{Events: ctx.EventManager().Events()}
 }
 
 func handleMsgAddCredential(ctx sdk.Context, k keeper.Keeper, msg types.MsgAddCredential) sdk.Result {
 	err := k.AddCredentials(ctx, msg.DidCredential.Claim.Id, msg.DidCredential)
 	if err != nil {
 		return err.Result()
 	}
-
-	return sdk.Result{}
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			types.EventTypeAddCredential,
+			sdk.NewAttribute(types.AttributeKeyCredType, msg.DidCredential.CredType[0]),
```

Comment:
Here we want to output the entire slice, not just the first item.

Reply:
ok. Noted and added in the similar way